chat-widget/src/components/livechatwidget/common/renderSurveyHelpers.ts

```diff
@@ -0,0 +1,117 @@
+import { Dispatch } from "react";
+import { Constants, ParticipantType } from "../../../common/Constants";
+import { LogLevel, TelemetryEvent } from "../../../common/telemetry/TelemetryConstants";
+import { TelemetryHelper } from "../../../common/telemetry/TelemetryHelper";
+import { addDelayInMs } from "../../../common/utils";
+import { ConversationState } from "../../../contexts/common/ConversationState";
+import { ILiveChatWidgetAction } from "../../../contexts/common/ILiveChatWidgetAction";
+import { ILiveChatWidgetContext } from "../../../contexts/common/ILiveChatWidgetContext";
+import { LiveChatWidgetActionType } from "../../../contexts/common/LiveChatWidgetActionType";
+import { PostChatSurveyMode } from "../../postchatsurveypanestateful/enums/PostChatSurveyMode";
+import { ILiveChatWidgetProps } from "../interfaces/ILiveChatWidgetProps";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let conversationDetails: any = undefined;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let postChatSurveyMode: any = undefined;
+
+const getBotSurveyMode = (props: ILiveChatWidgetProps, state: ILiveChatWidgetContext) => {
+    return (props.chatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveybotsurveymode ??
+        state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveybotsurveymode);
+};
+
+const getUserSurveyMode = (props: ILiveChatWidgetProps, state: ILiveChatWidgetContext) => {
+    return (props.chatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveymode ??
+        state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveymode);
+};
+
+// Set Survey mode based on conversation ended by entity
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const setSurveyMode = async (props: ILiveChatWidgetProps, participantType: string, state: ILiveChatWidgetContext, dispatch: Dispatch<ILiveChatWidgetAction>) => {
+    if (participantType === ParticipantType.User) {
+        postChatSurveyMode = getUserSurveyMode(props, state);
+        dispatch({ type: LiveChatWidgetActionType.SET_SURVEY_MODE, payload: postChatSurveyMode });
+        return;
+    }
+
+    if (participantType === ParticipantType.Bot) {
+        postChatSurveyMode = getBotSurveyMode(props, state);
+        dispatch({ type: LiveChatWidgetActionType.SET_SURVEY_MODE, payload: postChatSurveyMode });
+        return;
+    }
```

Comment:
Shall we default to User Mode if participant type is empty for some reason?

Reply:
Don't want to show black screen if getConversationDetails SDK method returns error